homeassistant/components/homekit/type_cameras.py

```diff
@@ -0,0 +1,211 @@
+                    VIDEO_CODEC_PARAM_LEVEL_TYPES["TYPE3_2"],
+                    VIDEO_CODEC_PARAM_LEVEL_TYPES["TYPE4_0"],
+                ],
+            },
+            "resolutions": resolutions,
+        }
+        audio_options = {"codecs": [{"type": "OPUS", "samplerate": 24}]}
+
+        stream_address = config_w_defaults.get(CONF_STREAM_ADDRESS, get_local_ip())
+
+        options = {
+            "video": video_options,
+            "audio": audio_options,
+            "address": stream_address,
+            "srtp": True,
+        }
+
+        super().__init__(
+            hass,
+            driver,
+            name,
+            entity_id,
+            aid,
+            config_w_defaults,
+            category=CATEGORY_CAMERA,
+            options=options,
+        )
+
+    def update_state(self, new_state):
+        """Handle state change to update HomeKit value."""
+        pass  # pylint: disable=unnecessary-pass
+
+    async def _async_get_stream_source(self):
+        """Find the camera stream source url."""
+        camera = self._camera.get_entity(self.entity_id)
+        if not camera or not camera.is_on:
+            return None
+        stream_source = self.config.get(CONF_STREAM_SOURCE)
+        if stream_source:
+            return stream_source
+        try:
+            return await camera.stream_source()
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception(
+                "Failed to get stream source - this could be a transient error or your camera might not be compatible with HomeKit yet"
+            )
+
+    async def start_stream(self, session_info, stream_config):
+        """Start a new stream with the given configuration."""
+        input_source = await self._async_get_stream_source()
+        if not input_source:
+            _LOGGER.error("Camera has no stream source")
+            return False
+
+        if "-i " not in input_source:
+            input_source = "-i " + input_source
+
+        output_vars = stream_config.copy()
+        output_vars.update(
+            {
+                "v_profile": VIDEO_PROFILE_NAMES[
+                    int.from_bytes(stream_config["v_profile_id"], byteorder="big")
+                ],
+                "v_bufsize": stream_config["v_max_bitrate"] * 2,
+                "v_map": self.config[CONF_VIDEO_MAP],
+                "v_pkt_size": self.config[CONF_VIDEO_PACKET_SIZE],
+                "a_bufsize": stream_config["a_max_bitrate"] * 2,
+                "a_map": self.config[CONF_AUDIO_MAP],
+                "a_pkt_size": self.config[CONF_AUDIO_PACKET_SIZE],
+                "a_encoder": AUDIO_ENCODER_OPUS,
+            }
+        )
+
+        self.start_stream_cmd = f"{self._ffmpeg.binary} {input_source} {VIDEO_OUTPUT}"
+        if self.config[CONF_SUPPORT_AUDIO]:
+            self.start_stream_cmd += " " + AUDIO_OUTPUT
+
+        return await super().start_stream(session_info, output_vars)
```

Comment:
Super calls `asyncio.create_subprocess_exec` which breaks with `Cannot add child handler, the child watcher does not have a loop attached` for me. I think that has something to do with how hass is started? (I'm just using `hass -c ./config` on macOS) There might be a way to make sure a loop is attached to the default child watcher? I'm unfamiliar with how all that works.

Reply:
Looks like we are going to have to go back to the custom start stream and stop stream.   That’s going to need more tests as well.

Reply:
@xdissent I reverted the change to user the super.  I'll work on a test later today.  Can you confirm the revert has you back to a good state?